### ISSUE TITLE
Could you describe how to perform gradient accumulation based on Detectron2?

### ISSUE THREAD

**Issue by HDUyiming** [justinkay/aldi#67]
I found that your work uses gradient accumulation, which enables a single card to run a large number of batchsize. However, I don't know exactly where your changes are. Could you please point out the location of the changes so that I can migrate this method to other projects based on Detectron2.

**Comment by justinkay** [comment]
Hi @HDUyiming , the gradient accumulation occurs here: https://github.com/justinkay/aldi/blob/2cf0a8dd6b96f7e3ee5d70dcdfd089afa97fd56a/aldi/trainer.py#L99 

Several forward passes are done (for weak & strong augmentations, source & target) before an optimizer step is taken.

**Comment by HDUyiming** [comment]
> Hi [@HDUyiming](https://github.com/HDUyiming) , the gradient accumulation occurs here:
> 
> [aldi/aldi/trainer.py](https://github.com/justinkay/aldi/blob/2cf0a8dd6b96f7e3ee5d70dcdfd089afa97fd56a/aldi/trainer.py#L99)
> 
> Line 99 in [2cf0a8d](/justinkay/aldi/commit/2cf0a8dd6b96f7e3ee5d70dcdfd089afa97fd56a)
> 
>  # Weakly-augmented source imagery (Used for normal training and/or domain alignment) 
> Several forward passes are done (for weak & strong augmentations, source & target) before an optimizer step is taken.

Thanks for you quickly reply!
